util.js

```diff
@@ -236,13 +236,20 @@
         customFavicon = domain.match(faviconDomains);
 
         if (customFavicon) {
-            // check for dark background & available variant
-            if (isDarkBg && darkVariants.test(customFavicon)) {
-                variant = '.white';
+            // validate domain
+            customFavicon = customFavicon.toString();
+            tokens = sourceUrl.split('.');
+            domainMatch = (tokens.length > 1) ? (tokens[tokens.length - 2] === customFavicon) : true;
+          
+            if (domainMatch) {
+                // check for dark background & available variant
+                if (DDG.settings && DDG.settings.get('kae') === 'd' && darkVariants.test(customFavicon)) {
```

Comment:
You're gonna hate this, but this doesn't work on the terminal theme ;)

I think there used to be an `isDark` property exposed on the updater, not sure what happened to it. It's still defined internally, should be pretty easy to re-expose again.